pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/MultiClusterHelixBrokerStarter.java

```diff
@@ -0,0 +1,358 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.helix.HelixConstants.ChangeType;
+import org.apache.helix.HelixManager;
+import org.apache.helix.HelixManagerFactory;
+import org.apache.helix.InstanceType;
+import org.apache.pinot.broker.routing.manager.MultiClusterRoutingManager;
+import org.apache.pinot.broker.routing.manager.RemoteClusterBrokerRoutingManager;
+import org.apache.pinot.common.config.provider.TableCache;
+import org.apache.pinot.common.config.provider.ZkTableCache;
+import org.apache.pinot.core.routing.MultiClusterRoutingContext;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.apache.pinot.spi.utils.CommonConstants.Helix;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Multi-cluster broker starter that extends the base Helix broker functionality
+ * to support federation across multiple Pinot clusters.
+ *
+ * This class handles:
+ * - Connection to remote clusters via separate ZooKeeper instances
+ * - Federated routing across primary and remote clusters
+ * - Cross-cluster query federation
+ * - Cluster change monitoring for remote clusters
+ */
+@SuppressWarnings("unused")
+public class MultiClusterHelixBrokerStarter extends BaseBrokerStarter {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MultiClusterHelixBrokerStarter.class);
+
+  // Remote cluster configuration
+  protected List<String> _remoteClusterNames;
+  protected Map<String, String> _remoteZkServers;
+  protected String _remoteInstanceId;
+
+  // Remote cluster Helix managers and routing
+  protected Map<String, HelixManager> _remoteSpectatorHelixManager;
+  protected Map<String, RemoteClusterBrokerRoutingManager> _remoteRoutingManagers;
+  protected MultiClusterRoutingManager _multiClusterRoutingManager;
+  protected MultiClusterRoutingContext _multiClusterRoutingContext;
+  protected Map<String, ClusterChangeMediator> _remoteClusterChangeMediator;
+
+  public MultiClusterHelixBrokerStarter() {
+  }
+
+  @Override
+  public void init(PinotConfiguration brokerConf)
+      throws Exception {
+    super.init(brokerConf);
+    _remoteInstanceId = _instanceId + "_remote";
+    initRemoteClusterNamesAndZk(brokerConf);
+  }
+
+  @Override
+  public void start()
+      throws Exception {
+    LOGGER.info("[multi-cluster] Starting multi-cluster broker");
+    super.start();
+    // build routing tables for remote clusters
+    initRemoteClusterRouting();
+    LOGGER.info("[multi-cluster] Multi-cluster broker started successfully");
+  }
+
+  @Override
+  protected void initSpectatorHelixManager() throws Exception {
+    super.initSpectatorHelixManager();
+    try {
+      initRemoteClusterSpectatorHelixManagers();
+    } catch (Exception e) {
+      LOGGER.error("[multi-cluster] Failed to initialize remote cluster spectator Helix managers", e);
```

Comment:
Let's emit a metric for `multi_cluster_broker_startup_failed`? We can re-use that counter at multiple places and rely on logs to find the exact root-cause.